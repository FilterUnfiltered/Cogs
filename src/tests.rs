#[test]
fn test_cogs() {
    let _ = crate::init_tracing();
    let tests = 1..=1;
    for test_index in tests {
        let name = format!("tests/{}.cog", test_index);
        let file = std::fs::read_to_string(&name).unwrap();
        let ast = crate::parse_cog(file, &name).unwrap();
        insta::with_settings!({ snapshot_suffix => format!("{test_index}") }, {
            insta::assert_debug_snapshot!(ast);
            insta::assert_snapshot!("codegen", cogs_codegen::generate(&ast).unwrap());
        });
    }
}
