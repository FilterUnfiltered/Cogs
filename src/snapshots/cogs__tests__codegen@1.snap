---
source: src/tests.rs
expression: "cogs_codegen::generate(&ast).unwrap()"
snapshot_kind: text
---
pub struct Cog;

impl cogs_runtime::Component for Cog {
    type Props = ();
    type Error = core::convert::Infallible;
    fn render(&self, props: Self::Props) -> impl core::future::Future<Output = Result<String, Self::Error>> + core::marker::Send + '_ {
        async move {
            let x = 1;let __rendered = format!(r#"<{}><{}>{}</{}><{} {}="{}">{}{}</{}>{}{}</{}>{}"#, r#"body"#,r#"h1"#,r#"Yo."#,r#"h1"#,r#"a"#,r#"src"#,r#"https://www.youtube.com/watch?v=dQw4w9WgXcQ"#,r#"Click this"#,x,r#"a"#,{println!("test");
format!(r#"<{}>{}</{}>"#, r#"p"#,r#"More Html"#,r#"p"#)
},r#""#,r#"body"#,r#""#);
            Ok(__rendered)
        }
    }
}
